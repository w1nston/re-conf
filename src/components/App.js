import React from 'react';
import { branch, compose, mapProps, renderComponent } from 'recompose';
import { gql, graphql } from 'react-apollo';
import Loading from '../components/Loading';
import Schedule from '../components/Schedule';
import parse from 'date-fns/parse';
import getDate from 'date-fns/get_date';
import addHours from 'date-fns/add_hours';

const withLoading = branch(
  props => props.data.loading,
  renderComponent(Loading)
);

const query = gql`
  query conferenceSchedule {
    events(slug: "reacteurope-2017") {
      schedule {
        id,
        title
        description
        startDate
        type
        length
        speakers {
          id,
          name
          avatarUrl
          twitter,
        }
      }
    }
  }
`;

const extractEvents = mapProps(props => ({ events: props.data.events }));
const extractSchedule = mapProps(props => ({
  schedule: props.events[0].schedule,
}));
const transformSchedule = mapProps(props => ({
  schedule: props.schedule.reduce((acc, event) => {
    const day = getDate(addHours(parse(event.startDate), 2));
    if (acc[day]) {
      acc[day].push(event);
    } else {
      acc[day] = [event];
    }
    return acc;
  }, {}),
}));

const ascending = (x, y) => {
  if (x.startDate < y.startDate) return -1;
  if (x.startDate > y.startDate) return 1;
  return 0;
};

const sortSchedule = mapProps(props => ({
  schedule: Object.keys(props.schedule).reduce((acc, day) => {
    acc[day] = props.schedule[day].sort(ascending);
    return acc;
  }, {}),
}));

const App = ({ schedule }) => <Schedule schedule={schedule} />;

export default compose(
  graphql(query),
  withLoading,
  extractEvents,
  extractSchedule,
  transformSchedule,
  sortSchedule
)(App);
