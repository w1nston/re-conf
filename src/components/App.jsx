import React from 'react';
import {
  BrowserRouter as Router,
  Route,
} from 'react-router-dom';
import Schedule from '../queries/Schedule';

export default function App() {
  return (
    <Router>
      <Route exact path="/" component={Schedule} />
    </Router>
  );
}
