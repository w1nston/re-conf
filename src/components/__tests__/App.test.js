import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('<App/>', () => {
  xit('renders', () => {
    // TODO: Figure out how to test Apollo containers
    const component = shallow(<App />);
  });
});
