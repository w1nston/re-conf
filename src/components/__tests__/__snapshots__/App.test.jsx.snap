exports[`<App/> renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <BrowserRouter>
    <Route
        component={[Function]}
        exact={true}
        path="/" />
</BrowserRouter>,
  "nodes": Array [
    <BrowserRouter>
      <Route
            component={[Function]}
            exact={true}
            path="/" />
</BrowserRouter>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <BrowserRouter>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/" />
</BrowserRouter>,
        "_debugID": 2,
        "_renderedOutput": <BrowserRouter>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/" />
</BrowserRouter>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
